### DISCUSSION TITLE
Impact in large number of coercion rules

### DISCUSSION THREAD

**Discussion by albertosaez** [dcm4che/dcm4chee-arc-light#2919] [Q&A] [answered]
In my scenario, I have a large number of modalities, and for most of them I need to configure 2 coercion: one for the queries to the worklist and other for the storage.

This would mean the definition of > 400 coercion rules. Does this have an impact in the system performance ?

The coercion for the worklist is for adding a filter  based on aetitle, and the ones for the storage are for setting Institution and department.

**Comment by gunterze** [comment, marked as answer]
Why do you think you need individual coercion rules for each modality?

You just need **one** rule with [**Use Calling AE Title as:**](https://dcm4chee-arc-cs.readthedocs.io/en/latest/networking/config/archiveAttributeCoercion.html#dcmusecallingaetitleas) `ScheduledStationAETitle` to set _Scheduled Station AE Title (0040,0001)_ in item of _Scheduled Procedure Step Sequence (0040,0100)_ of C-FIND RQs from **all** modalities to the _Calling AE Title_ they used to initiate the Association.

And you need only one rule by different value for _Institutional Department Name (0008,1040)_ you want to supplement in receiving C-STORE RQs, selected by **Condition:** `SendingApplicationEntityTitle=MOD1|MOD2|MOD3` or by configuring individual Archive AEs for each different Department (e.g. with AE Titles: `4DEPT1`, `4DEPT2`, ...) and configuring the rule on Archive Network AE level - which would also allow to configure different 

Name | Type | Description (LDAP Attribute)
-- | -- | --
Store Access Control ID | string | Access Control ID assigned to Studies received by this AE (dcmStoreAccessControlID)

if not every client shall have access to all studies independently from its source. s. [Access control based on Archive AEs](https://github.com/dcm4che/dcm4chee-arc-light/wiki/Access-control-based-on-Archive-AEs)

Supplementing can be either configured directly by [Archive Attribute Coercion Attribute](https://dcm4chee-arc-cs.readthedocs.io/en/latest/networking/config/archiveAttributeCoercion.html)

Name | Type | Description (LDAP Attribute)
-- | -- | --
Merge Attribute(s) | string | Merge DICOM Attribute in format  {attributeID}={value}. {attributeID} inside of {value} will be replaced  by the value of that attribute in the original dataset. Example:  InstitutionName=RegionalHospital or 00081040=XRay. (dcmMergeAttribute)

or by referring another Device (e.g.  `dept1`, `dept2`, ...) by

Name | Type | Description (LDAP Attribute)
-- | -- | --
Supplement from Device | string | Name of Device from which Assigning Authorities and  other information is taken to supplement received Composite Objects and  MPPS. (dcmSupplementFromDeviceName)

with configured [Device Attributes](https://dcm4chee-arc-cs.readthedocs.io/en/latest/networking/config/device.html)

Name | Type | Description (LDAP Attribute)
-- | -- | --
Institution Name(s) | string | Should be the same as the value of Institution Name (0008,0080) in SOP Instances created by this device (dicomInstitutionName)
Institution Department Name(s) | string | Should be the same as the value of Institutional Department Name (0008,1040) in SOP Instances created by this device (dicomInstitutionDepartmentName)

s.o. [Supplement received MPPS and DICOM Objects with Attributes](https://github.com/dcm4che/dcm4chee-arc-light/wiki/Supplement-received-MPPS-and-DICOM-Objects-with-Attributes)

Anyway,  I also  guess, that iterating through >400 coercion rules should not significantly harm performance.
